pkg/collector/bucket_collector.go

```diff
@@ -0,0 +1,87 @@
+package collector
+
+import (
+	"context"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"gocloud.dev/blob"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// BucketCollector stores the state for the git collector
+type BucketCollector struct {
+	bucketURL  string
+	bucket     *blob.Bucket
+	classifier string
+}
+
+// NewBucketCollector creates a new git based collector
+func NewBucketCollector(bucketURL string, bucket *blob.Bucket, classifier string) (Collector, error) {
+	return &BucketCollector{
+		bucketURL:  bucketURL,
+		bucket:     bucket,
+		classifier: classifier,
+	}, nil
+}
+
+// CollectFiles collects files and returns the URLs
+func (c *BucketCollector) CollectFiles(patterns []string, outputPath string, basedir string) ([]string, error) {
+	urls := []string{}
+	bucket := c.bucket
+
+	ctx := context.Background()
```

Comment:
nit: Maybe consider creating the context with a timeout to avoid cases when it hangs forever.

`context.WithTimeout(context.Background(), timeout)`